docs/learn/prompts.md

```diff
@@ -333,212 +235,48 @@ print(prompt.message_params())
-    """Recommend a {genre} book"""
-
-    genre: str
-
-
-prompt = BookRecommendationPrompt(genre="fantasy")
-print(prompt)
-# > Recommend a fantasy book
+## Best Practices
 
+To make the most of Mirascope's prompt features, consider the following best practices:
 
-@openai.call(model="gpt-4o-mini")
-def recommend_book(genre: str):
-    """Recommend a {genre} book"""
-    ...
+- **Prompt Libraries**: Build libraries of commonly used prompts that can be easily shared across projects or teams.
+   ```python
+   # prompts/book_recommendations.py
+   @prompt_template("Recommend a {genre} book by {author}")
+   def book_recommendation_prompt(genre: str, author: str): ...
 
+   # In your main code
+   from prompts.book_recommendations import author_based_recommendation
+   ```
 
-response = recommend_book("mystery")
-print(response.content)
-# > Here's a recommendation for a fantasy book: ...
-```
+- **Testing Your Prompts**: Create unit tests for your prompts to ensure they generate the expected output.
+   ```python
+   def test_book_recommendation_prompt():
+       prompt = book_recommendation_prompt("mystery")
+       assert "Recommend a mystery book" in prompt[0].content
+   ```
 
-!!! warning
+For more advanced features like computed fields or running prompts with different providers, see the respective sections in our documentation:
 
-    Using docstrings as prompt templates can make your code less explicit and harder to maintain. It's generally recommended to use the `@prompt_template` decorator for clarity and separation of concerns. Enable this feature at your own risk.
-
-## Best Practices
+- [Dynamic Configuration](./dynamic_configuration.md)
+- [Calls](./calls.md)
+- [Response Models](./response_models.md)
 
-- Provider Comparison: Use `BasePrompt` to easily test the same prompt across different providers to compare performance and output quality.
-- Prompt Versioning: Utilize the metadata decorator to keep track of different versions of your prompts as you refine them.
-- Dynamic Content: Leverage `@computed_field` for injecting dynamic content or API calls into your prompts.
-- Cached Properties: Use `@functools.cached_property` to cache frequently used properties that you only want to compute once.
-- Prompt Libraries: Build libraries of commonly used prompts that can be easily shared across projects or teams.
+By mastering prompts in Mirascope, you'll be well-equipped to build robust, flexible, and reusable LLM applications.
 
-Mastering `BasePrompt` is the first step towards building robust LLM applications with Mirascope that are flexible, reusable, and provider-agnostic.
+!!! note "Deprecation of BasePrompt"
+    We are in the process of deprecating the `BasePrompt` class in favor of the more functional approach shown in this documentation. This change will simplify maintenance and provide a more consistent API. For more information on upcoming changes, see our [issue tracker](https://github.com/Mirascope/mirascope/issues/490).
```

Comment:
@willbakst 
I don't know you prefer the section. But, I added it to avoid confusion of user when user see `BasePrompt` example.
What do you think about it?

Reply:
which `BasePrompt` example?

I think the better thing to do would be to add a deprecation warning on the class and then just remove all examples and documentation that uses it?